### PR TITLE
geo/geos: refactor C++ initialization code

### PR DESCRIPTION
**Pull request by petermattis**
Refactor the C++ initialization code, and remove the `CR_GEOS_` from
member names. The latter seems to fix a warning that some C++ compilers
complain about.

Release note: None

### CODE REVIEW

**Review by otan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan, @petermattis, and @sumeerbhola)

___
*[pkg/geo/geos/geos_unix.cc, line 45 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4-ppzl1ggAFbbWxAYv:-M4-ppzm6zd3SJc70NoP:b-p11bq) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d33d44abe755effcf31b9cfd2d85274b530392a/pkg/geo/geos/geos_unix.cc#L45)):*
> ```cpp
>   void *dlHandle;
> 
>   CR_GEOS_init_r CR_GEOS_init_r;
> ```

i'm thinking the variable name should still be `GEOS_finish_r` to match the GEOS name for them.

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @sumeerbhola)

___
*[pkg/geo/geos/geos_unix.cc, line 45 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4-ppzl1ggAFbbWxAYv:-M4-qjWGBxUR8gnnZM1O:btrvo7m) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d33d44abe755effcf31b9cfd2d85274b530392a/pkg/geo/geos/geos_unix.cc#L45)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

i'm thinking the variable name should still be `GEOS_finish_r` to match the GEOS name for them.
</blockquote></details>

How important is that? In C you'd do that because of lack of namespaces and scoping. But C++ doesn't have such restrictions. On the other hand, keeping the name consistent might ease grep-ability.

**Comment by otan** [comment]
___
*[pkg/geo/geos/geos_unix.cc, line 45 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4-ppzl1ggAFbbWxAYv:-M402RyM-3jNZWO4S8CZ:bikiaix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d33d44abe755effcf31b9cfd2d85274b530392a/pkg/geo/geos/geos_unix.cc#L45)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

How important is that? In C you'd do that because of lack of namespaces and scoping. But C++ doesn't have such restrictions. On the other hand, keeping the name consistent might easy grep-ability. 
</blockquote></details>

basically yeah i want it for the greppyness :)

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @sumeerbhola)

___
*[pkg/geo/geos/geos_unix.cc, line 45 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4-ppzl1ggAFbbWxAYv:-M4043Sl0gbrHwALWoZD:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d33d44abe755effcf31b9cfd2d85274b530392a/pkg/geo/geos/geos_unix.cc#L45)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

basically yeah i want it for the greppyness :)
</blockquote></details>

Done.
___
*[pkg/geo/geos/geos_unix.cc, line 86 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4045O3-mwbhisrxB3X:-M4045O3-mwbhisrxB3Y:bx7wd6t) ([raw file](https://github.com/cockroachdb/cockroach/blob/387b533c1fde732ae53a9355d1b49e25927e5dcb/pkg/geo/geos/geos_unix.cc#L86)):*
> ```cpp
>         return error;               \
>       }                             \
>     } while (0)
> ```

@otan Is this too clever? It has the side-effect of guaranteeing that the field names match the symbol names in libgeos.

**Comment by tbg** [comment]
Heh, of course you already had a PR open for something like this.

**Review by sumeerbhola — APPROVED** [review]
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geos/geos_unix.cc, line 86 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4045O3-mwbhisrxB3X:-M40PFEF6OHwy_iXP4gF:bb7lbs2) ([raw file](https://github.com/cockroachdb/cockroach/blob/387b533c1fde732ae53a9355d1b49e25927e5dcb/pkg/geo/geos/geos_unix.cc#L86)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

@otan Is this too clever? It has the side-effect of guaranteeing that the field names match the symbol names in libgeos.
</blockquote></details>

nice

**Review by sumeerbhola — DISMISSED** [review]
(didn't mean to hit approve -- otan is the real reviewer for this)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-:-M40PMeNCs_71FDdL5Ea:b7en98v)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

**Review by otan — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @petermattis)

___
*[pkg/geo/geos/geos_unix.cc, line 18 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M402fn3-C_ZCjorfq-h:-M402fn3-C_ZCjorfq-i:bonl1ha) ([raw file](https://github.com/cockroachdb/cockroach/blob/cfa168cae1fd974797600c052b1ca2421e696f06/pkg/geo/geos/geos_unix.cc#L18)):*
> ```cpp
> #include "geos_unix.h"
> 
> namespace {
> ```

mind also giving this a clang-format? want to see if namespaces need tabs underneath (seems hard to read otherwise, but then again so is `#ifndef _BLAH <slab> #endif`)
___
*[pkg/geo/geos/geos_unix.cc, line 86 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M4045O3-mwbhisrxB3X:-M40UnBw9W-7lcldDmlt:b-vcrl4q) ([raw file](https://github.com/cockroachdb/cockroach/blob/387b533c1fde732ae53a9355d1b49e25927e5dcb/pkg/geo/geos/geos_unix.cc#L86)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

nice
</blockquote></details>

that's a #define trick if i ever saw one :')
i like it, hah!

**Review by petermattis — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @petermattis)

___
*[pkg/geo/geos/geos_unix.cc, line 18 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-M402fn3-C_ZCjorfq-h:-M40X2rwBFMJ_AsFv0GL:b39ekqn) ([raw file](https://github.com/cockroachdb/cockroach/blob/cfa168cae1fd974797600c052b1ca2421e696f06/pkg/geo/geos/geos_unix.cc#L18)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

mind also giving this a clang-format? want to see if namespaces need tabs underneath (seems hard to read otherwise, but then again so is `#ifndef _BLAH <slab> #endif`)
</blockquote></details>

Done. I also copied over the `.clang-format` config from `c-deps/libroach`. All the reformatting is contained in a separate commit.

**Review by otan — APPROVED** [review]
looks good to me!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46984#-:-M40sARCDeiIhBKQnZBZ:bffd1li)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan, @petermattis, and @sumeerbhola)

**Comment by petermattis** [comment]
TFTR!

bors r+
